model/build/build.go

```diff
@@ -252,7 +252,7 @@ func TryMarkStarted(buildId string, startTime time.Time) error {
 func (b *Build) MarkFinished(status string, finishTime time.Time) error {
 	b.Status = status
 	b.FinishTime = finishTime
-	b.TimeTaken = finishTime.Sub(b.StartTime)
+	b.TimeTaken = finishTime.Sub(b.ActivatedTime)
```

Comment:
Sorry for the late CR -- the issue here is that the task can be activated before it actually is started, since it has to wait for a host, and that could be a while.

Reply:
In these cases where we perform restarts to be completely accurate we'd need the start time of the most recent run which I don't think is readily available and might need some more doing - the change to begin with the activation time in this case though would still definitely be an improvement over what is currently happening.